packages/model/src/data-providers.ts

```diff
@@ -16,26 +16,37 @@ export type FetchCalendarProviderOptions = {
   resourceId?: string;
 };
 
-export interface CalendarDataProvider<
-  CalendarDataObject = DefaultCalendarDataObject,
-  ListCalendarDataObject = DefaultListCalendarDataObject,
+export interface DataProvider<
+  TItemResponse = null,
+  TFetchOptions = null,
+  TListItemResponse = {
+    items: TItemResponse[];
+    total: number;
+  },
+  TCreateData = never,
```

Comment:
a nice alternative is to do this:
```
type DataProvider<T = {}, K = null>  = {
  fetchById: (id: string) => Promise<T | null>;
} & (K extends null ? {} : {
  create: K extends null ? undefined : (data: K) => Promise<string>;
});
```
you can still implement regular Types and with the above you are able to conditionally add create or update based on the input generic. Ill leave it up to you to decide which one is better